mypy/checker.py

```diff
@@ -1040,8 +1055,9 @@ def check___new___signature(self, fdef: FuncDef, typ: CallableType) -> None:
         self_type = fill_typevars_with_any(fdef.info)
         bound_type = bind_self(typ, self_type, is_classmethod=True)
         # Check that __new__ (after binding cls) returns an instance
-        # type (or any)
-        if not isinstance(bound_type.ret_type, (AnyType, Instance, TupleType)):
+        # type (or any).
+        if not isinstance(get_proper_type(bound_type.ret_type),
```

Comment:
> Perhaps this can come later as we convert our various helper functions to work with ProperType instead of Type?

Yes, we can try to put more order here after get some experience, currently it is pure intuition.